js/apps/admin-ui/src/user/EditUser.tsx

```diff
@@ -203,14 +203,17 @@ export default function EditUser() {
           </DropdownItem>,
         ]}
         onToggle={(value) =>
-          save({ ...toUserFormFields(user), enabled: value })
+          save({
+            ...toUserFormFields(user, !!userProfileMetadata),
+            enabled: value,
+          })
         }
         isEnabled={user.enabled}
       />
 
       <PageSection variant="light" className="pf-u-p-0">
         <UserProfileProvider>
-          <FormProvider {...userForm}>
+          <FormProvider {...form}>
```

Comment:
Either we use the form as a parameter or we pass it in as an argument, here you do both. IMO using the `FormProvider` is a bit nicer

Reply:
> Either we use the form as a parameter or we pass it in as an argument

Agreed, but I did not want to refactor all the forms under this provider in this PR as that would create significantly more code to review.

 > IMO using the `FormProvider` is a bit nicer

I agree with you from a convenience perspective, it is a lot less verbose to use a context. However I disagree from a type-safety and maintenance perspective. The reason for that is when you use a context the form and it's value becomes loosely coupled from the form controls.

For example, I can do the following that is perfectly valid:


```tsx
type ParentForm = {
  attributes: Record<string, string[]>;
}

const ParentWithForm =  () => {
  const form = useForm();
  return <FormProvider {...form}><ChildWithContext /></FormProvider>
}


type ChildForm = {
  attributes: { key: string; value: string; }[];
}

const ChildWithContext =  {
  const form = useFormContext();
  // This will not give an error, even thought it is wrong.
  // The parent form is expecting a different type.
  form.setValue("attributes", [{ key: "foo", value: "bar" }]);
}
```

If we pass the form down as a strongly typed prop instead, then it will suddenly not be possible to assign it to anything but the type that was specified at the root. Also it will prevent possible nested `FormProvider`s, which can cause conflicts with each other.

So yeah, it  is more cumbersome, but it is also more maintainable long term, and I think that is more important than the DX in the shot term.

Reply:
that's why you should use `useFormContext<ParentForm>();`, but yes you can remove the types, but that is true everywhere because in the end there are no types.

Reply:
This has been resolved as we now pass the form in as a typed prop everywhere for User Profile.

Reply:
If that is the case why do you still have this `<FormProvider>` tag here

Reply:
Ah yes, my bad. I meant for the user profile fields. Doing all the other fields for the user would be a massive refactor that I don't want to get into right now.